brvs-core/src/main/java/iroha/validation/listener/IrohaReliableChainListener.java

```diff
@@ -0,0 +1,185 @@
+package iroha.validation.listener;
+
+import com.google.common.base.Strings;
+import com.google.common.util.concurrent.Runnables;
+import com.rabbitmq.client.BuiltinExchangeType;
+import com.rabbitmq.client.CancelCallback;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.DeliverCallback;
+import com.rabbitmq.client.Delivery;
+import com.rabbitmq.client.MessageProperties;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+import iroha.protocol.BlockOuterClass;
+import iroha.protocol.BlockOuterClass.Block;
+import iroha.protocol.QryResponses;
+import iroha.protocol.Queries;
+import iroha.protocol.TransactionOuterClass;
+import iroha.protocol.TransactionOuterClass.Transaction;
+import java.io.Closeable;
+import java.io.IOException;
+import java.security.KeyPair;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import jp.co.soramitsu.iroha.java.BlocksQueryBuilder;
+import jp.co.soramitsu.iroha.java.IrohaAPI;
+import jp.co.soramitsu.iroha.java.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class IrohaReliableChainListener implements Closeable {
+
+  private static final Logger logger = LoggerFactory.getLogger(IrohaReliableChainListener.class);
+
+  private final IrohaAPI irohaAPI;
+  // BRVS account id to query Iroha
+  private final String accountId;
+  // BRVS keypair to query Iroha
+  private final KeyPair keyPair;
+
+  private static final String EXCHANGE_RMQ_NAME = "iroha";
+
+  private Connection connection;
+  private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
+
+  private boolean isListening;
+
+  @Autowired
+  public IrohaReliableChainListener(
+      IrohaAPI irohaAPI,
+      String accountId,
+      KeyPair keyPair,
+      String RMQHost,
```

Comment:
we don't call variables like this in Java.

Reply:
Done